File: ankor-core/src/main/java/at/irian/ankor/messaging/MessageMapper.java
package at.irian.ankor.messaging;

/**
 * A MessageMapper is responsible of serializing and deserializing messages for the transport layer.
 * @param <T>  type of serialized message
 * @author Manfred Geiler
 */
public interface MessageMapper<T> extends MessageSerializer<T>, MessageDeserializer<T> {
}

File: #ANKOR.net/Ankor.Core/Messaging/MessageMapper.cs
namespace Ankor.Core.Messaging {
	/// <summary>
	/// Description of MessageMapper.
	/// </summary>
	public interface MessageMapper<T> : MessageSerializer<T>, MessageDeserializer<T>  {
		

	}
	
	public interface MessageSerializer<T> {
		T Serialize(object msg);
	}
	
	public interface MessageDeserializer<T> {
		TR Deserialize<TR>(T serializedMsg);
	}
}
